plone/app/event/base.py

```diff
@@ -279,10 +279,9 @@ def expand_events(events, ret_mode,
     for it in events:
         obj = it.getObject() if getattr(it, 'getObject', False) else it
         if IEventRecurrence.providedBy(obj):
-
             occ_list = list(IRecurrenceSupport(obj).occurrences(start, end))
             # add original event (to make expand_events not remove one item of the original list)
-            if (start or end) and (IEventAccessor(obj).start >= (end or start)):
+            if (start or end) and (obj.start >= (end or start)):
```

Comment:
I think you have to take the whole_day setting also into account. That's why IEventAccessor is useful for. See: https://github.com/plone/plone.app.event/blob/f6302dd7444b6037aa9f53ebb0779acf70246601/plone/app/event/dx/behaviors.py#L469
Probably the ``_recurrence_upcoming_event`` is a problem for using ``IEventAccessor``: https://github.com/plone/plone.app.event/blob/f6302dd7444b6037aa9f53ebb0779acf70246601/plone/app/event/dx/behaviors.py#L464
That ``_recurrence_upcoming_event`` probably should be a separate function, but I don't know yet.